src/context.jl

```diff
@@ -34,11 +34,6 @@ Behavior can be overridden by setting the `CLIMACOMMS_CONTEXT` environment varia
 to either `MPI` or `SINGLETON`.
 """
 function context(device = device(); target_context = context_type())
-    if target_context == :MPICommsContext && mpi_ext_is_not_loaded()
-        error(
-            "Loading MPI.jl is required to use MPICommsContext. You might want to call ClimaComms.@import_required_backends",
-        )
-    end
```

Comment:
I think we should produce an informative error when we know that the issue is that the extension was not loaded

Reply:
Can we instead just add some documentation about this? I'd prefer we avoid putting code loading logic into a function just to return a (slightly) more helpful error message.

Reply:
Ah, just saw your other comment. Perhaps you're right. Let's see if we can get away with a patch release.

Reply:
> Can we instead just add some documentation about this? I'd prefer we avoid putting code loading logic into a function just to return a (slightly) more helpful error message.

This function does no code loading. It only checks types. In my opinion this error message is much more useful than any documentation. What this error message does is this:

Suppose you are in a SLRUM job (where context is automatically MPI) and run this script:
```
import ClimaComms
ClimaComms.context()
```
Without this check, you would get something like `No methods exist for MPICommsContext`. Now, form here, users would have to figure out that this is not a real issue, but they just forgot to load MPI.jl. Instead, the error message is very clear, even those that know nothing about extensions will immediately know what to do next (we will see if the instruction is to load MPI.jl or to use required_backends, but that's a detail).

Reply:
`mpi_ext_is_not_loaded` makes this a stateful function based on two variables which, IMO, is far more complicated than being a stateful function based on `ENV` alone. Let's move forward with good documentation and simpler code.

Reply:
`ClimaComms.context()` is one of those functions that everyone uses. We are in the position where we can almost prevent an entire class of problems from occuring by precisely telling users what went wrong and what they need to do (educating them in the process). In my opinion, this is worth much more than having a function that depends on the state of two variables. No one reads the documentation for ClimaComms and people won't even know where to start looking when they are hit with a `MPICommsContext has no methods` error. 

We can ask someone else to weigh in their opinion.